### ISSUE TITLE
Setting avoidExternalConnections breaking url pattern transformation and browser driver download.

### ISSUE THREAD

**Issue by abulat** [bonigarcia/webdrivermanager#1156]
**Description of the problem**: 
I cannot download the latest browser driver version after updating to a new *webdrivermanager* *v5.6.0* and starting using a avoidExternalConnections option. 

I checked in both cases with using my custom nexus pattern and default coming with webdrivermanager lib.

**Browser and version**:
Chrome

**Operating system and architecture**: 
Windows 64

**Selenium version**: 
118.0.5993.70 

**WebDriverManager version**: 
5.6.0

**WebDriverManager call**: 
```WebDriverManager.chromedriver().avoidExternalConnections().setup();```

**WebDriverManager traces**: 
N/A

**Error log**: 
```
2023-10-31 14:01:01:568 +0100 [main] INFO io.github.bonigarcia.wdm.WebDriverManager - Using chromedriver 118.0.5993.70 (resolved driver for Chrome 118)
2023-10-31 14:01:01:572 +0100 [main] INFO io.github.bonigarcia.wdm.cache.ResolutionCache - Clearing WebDriverManager resolution cache
2023-10-31 14:01:01:572 +0100 [main] WARN io.github.bonigarcia.wdm.WebDriverManager - There was an error managing chromedriver 118.0.5993.70 (Malformed escape pair at index 60: https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/%s/%s/chrom
edriver-%s.zip) ... trying again avoiding reading release from repository
2023-10-31 14:01:01:715 +0100 [main] WARN io.github.bonigarcia.wdm.WebDriverManager - There was an error managing chromedriver (latest version) (Malformed escape pair at index 60: https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/%s/%s/ch
romedriver-%s.zip) ... trying again using latest driver stored in cache
2023-10-31 14:01:01:718 +0100 [main] ERROR io.github.bonigarcia.wdm.WebDriverManager - There was an error managing chromedriver (latest version) (Malformed escape pair at index 60: https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/%s/%s/c
hromedriver-%s.zip)
java.lang.IllegalArgumentException: Malformed escape pair at index 60: https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/%s/%s/chromedriver-%s.zip
        at java.base/java.net.URI.create(URI.java:883)
        at io.github.bonigarcia.wdm.WebDriverManager.download(WebDriverManager.java:1273)
        at io.github.bonigarcia.wdm.WebDriverManager.manage(WebDriverManager.java:1189)
        at io.github.bonigarcia.wdm.WebDriverManager.fallback(WebDriverManager.java:1445)
        at io.github.bonigarcia.wdm.WebDriverManager.handleException(WebDriverManager.java:1426)
        at io.github.bonigarcia.wdm.WebDriverManager.manage(WebDriverManager.java:1195)
        at io.github.bonigarcia.wdm.WebDriverManager.handleException(WebDriverManager.java:1419)
        at io.github.bonigarcia.wdm.WebDriverManager.manage(WebDriverManager.java:1195)
        at io.github.bonigarcia.wdm.WebDriverManager.setup(WebDriverManager.java:427)
```

**Comment by giulong** [comment]
The `avoidExternalConnections` property should be used in combination with `chromeDownloadUrlPattern`, I mean you need to set your internal Nexus url there. It's not clear to me from the info provided if you're currently setting it (at least in the provided API chain that's not set). Can you clarify?

---
Side note for @bonigarcia: I now notice I hardcoded a call to `getChromeDownloadUrlPattern` [in the generic WebDriverManager](https://github.com/bonigarcia/webdrivermanager/blob/625e28db2df3a2fae07e81b5f18bea994467be4b/src/main/java/io/github/bonigarcia/wdm/WebDriverManager.java#L1273). This is not affeccting this specific case, since the OP is using Chrome. I will work on a generic fix though.

**Comment by abulat** [comment]
>  I mean you need to set your internal Nexus url there. It's not clear to me from the info provided if you're currently setting it (at least in the provided API chain that's not set). Can you clarify?

@giulong I am getting the same exception if I set my nexus url pattern and if I don't set it.

**Comment by giulong** [comment]
will look into it. Keep you posted

**Comment by bonigarcia** [comment]
I have just released wdm 5.6.1, containing the fix by @giulong about this.
